From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Johannes Obermayr <johannesobermayr@gmx.de>
Date: Sat, 21 Jan 2023 23:16:51 +0100
Subject: [PATCH] 4910: Fix build error after 8b442ea

> Task :Decompiler:compileSleighLinux_x86_64ExecutableSleighCpp
/tmp/ghidra/Ghidra/Features/Decompiler/src/decompile/cpp/float.cc: In member function ???double FloatFormat::getHostFloat(uintb, floatclass*) const???:
/tmp/ghidra/Ghidra/Features/Decompiler/src/decompile/cpp/float.cc:245:30: error: ???numeric_limits??? is not a member of ???std???
  245 |       double infinity = std::numeric_limits<double>::infinity();
      |                              ^~~~~~~~~~~~~~
/tmp/ghidra/Ghidra/Features/Decompiler/src/decompile/cpp/float.cc:245:45: error: expected primary-expression before ???double???
  245 |       double infinity = std::numeric_limits<double>::infinity();
      |                                             ^~~~~~
/tmp/ghidra/Ghidra/Features/Decompiler/src/decompile/cpp/float.cc:250:23: error: ???numeric_limits??? is not a member of ???std???
  250 |     double nan = std::numeric_limits<double>::quiet_NaN();
      |                       ^~~~~~~~~~~~~~
/tmp/ghidra/Ghidra/Features/Decompiler/src/decompile/cpp/float.cc:250:38: error: expected primary-expression before ???double???
  250 |     double nan = std::numeric_limits<double>::quiet_NaN();
      |                                      ^~~~~~

> Task :Decompiler:compileDecompileLinux_x86_64ExecutableDecompileCpp
/tmp/ghidra/Ghidra/Features/Decompiler/src/decompile/cpp/float.cc: In member function ???double FloatFormat::getHostFloat(uintb, floatclass*) const???:
/tmp/ghidra/Ghidra/Features/Decompiler/src/decompile/cpp/float.cc:245:30: error: ???numeric_limits??? is not a member of ???std???
  245 |       double infinity = std::numeric_limits<double>::infinity();
      |                              ^~~~~~~~~~~~~~
/tmp/ghidra/Ghidra/Features/Decompiler/src/decompile/cpp/float.cc:245:45: error: expected primary-expression before ???double???
  245 |       double infinity = std::numeric_limits<double>::infinity();
      |                                             ^~~~~~
/tmp/ghidra/Ghidra/Features/Decompiler/src/decompile/cpp/float.cc:250:23: error: ???numeric_limits??? is not a member of ???std???
  250 |     double nan = std::numeric_limits<double>::quiet_NaN();
      |                       ^~~~~~~~~~~~~~
/tmp/ghidra/Ghidra/Features/Decompiler/src/decompile/cpp/float.cc:250:38: error: expected primary-expression before ???double???
  250 |     double nan = std::numeric_limits<double>::quiet_NaN();
      |                                      ^~~~~~
---
 Ghidra/Features/Decompiler/src/decompile/cpp/float.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/float.cc b/Ghidra/Features/Decompiler/src/decompile/cpp/float.cc
index 23bd6b385..38c973ce8 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/float.cc
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/float.cc
@@ -18,6 +18,7 @@
 #include "address.hh"
 
 #include <cmath>
+#include <limits>
 using std::ldexp;
 using std::frexp;
 using std::signbit;
-- 
2.39.0

